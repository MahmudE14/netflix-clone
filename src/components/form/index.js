import { Container, Base, Error, Title, Text, TextSmall, Link, Input, Submit } from './styles/form'

export default function Form({ children, ...restProps }) {
    return (
        <Container {...restProps}>{children}</Container>
    )
};

Form.Base = function FormContainer({ children, ...restProps }) {
    return <Base { ...restProps }>{children}</Base>
}

Form.Error = function FormContainer({ children, ...restProps }) {
    return <Error { ...restProps }>{children}</Error>
}

Form.Title = function FormContainer({ children, ...restProps }) {
    return <Title { ...restProps }>{children}</Title>
}

Form.Text = function FormContainer({ children, ...restProps }) {
    return <Text { ...restProps }>{children}</Text>
}

Form.TextSmall = function FormContainer({ children, ...restProps }) {
    return <TextSmall { ...restProps }>{children}</TextSmall>
}

Form.Link = function FormContainer({ children, ...restProps }) {
    return <Link { ...restProps }>{children}</Link>
}

Form.Input = function FormContainer({ children, ...restProps }) {
    return <Input { ...restProps }>{children}</Input>
}

Form.Submit = function FormContainer({ children, ...restProps }) {
    return <Submit { ...restProps }>{children}</Submit>
}
