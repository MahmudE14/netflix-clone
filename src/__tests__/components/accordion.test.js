import { fireEvent, render, screen } from "@testing-library/react";
import { Accordion } from "../../components";
import faqsData from "../../fixtures/faqs.json";

describe("<Accordion />", () => {
    it("renders <Accordion /> with populated data", () => {
        const { container, getByText } = render(
            <Accordion>
                <Accordion.Title>Frequently Asked Questions</Accordion.Title>
                {faqsData.map((item) => (
                    <Accordion.Item key={item.id}>
                        <Accordion.Header>{item.header}</Accordion.Header>
                        <Accordion.Body>{item.body}</Accordion.Body>
                    </Accordion.Item>
                ))}
            </Accordion>
        );

        expect(screen.getByText("Frequently Asked Questions")).toBeTruthy();
        expect(screen.getByText("What is Netflix?")).toBeTruthy();
        expect(screen.getByText("How much does Netflix cost?")).toBeTruthy();
        expect(screen.getByText("Where can I watch?")).toBeTruthy();
        expect(screen.getByText("How do I cancel?")).toBeTruthy();
        expect(screen.getByText("What can I watch on Netflix?")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it("opens and closes the <Accordion /> component", () => {
        const { container, getByText } = render(
            <Accordion>
                <Accordion.Title>Frequently Asked Questions</Accordion.Title>
                {faqsData.map((item) => (
                    <Accordion.Item key={item.id}>
                        <Accordion.Header>{item.header}</Accordion.Header>
                        <Accordion.Body>{item.body}</Accordion.Body>
                    </Accordion.Item>
                ))}
            </Accordion>
        );

        const whatIsNetflixDetails = "Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices. You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!";

        expect(screen.queryByText(whatIsNetflixDetails)).toBeFalsy()

        fireEvent.click(screen.queryByText('What is Netflix?'))
        expect(screen.getByText(whatIsNetflixDetails)).toBeTruthy()

        fireEvent.click(screen.queryByText('What is Netflix?'))
        expect(screen.queryByText(whatIsNetflixDetails)).toBeFalsy()

        expect(container.firstChild).toMatchSnapshot()
    });
});
