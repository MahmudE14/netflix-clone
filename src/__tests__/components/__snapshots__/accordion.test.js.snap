// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> opens and closes the <Accordion /> component 1`] = `
<div
  class="accordion__Container-sc-1wnyt6p-0 kRFsOa"
>
  <div
    class="accordion__Inner-sc-1wnyt6p-1 ifgAxd"
  >
    <h1
      class="accordion__Title-sc-1wnyt6p-2 dUpgvE"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-sc-1wnyt6p-3 iMUxK"
    >
      <div
        class="accordion__Header-sc-1wnyt6p-4 fDhNEC"
      >
        What is Netflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1wnyt6p-3 iMUxK"
    >
      <div
        class="accordion__Header-sc-1wnyt6p-4 fDhNEC"
      >
        How much does Netflix cost?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1wnyt6p-3 iMUxK"
    >
      <div
        class="accordion__Header-sc-1wnyt6p-4 fDhNEC"
      >
        Where can I watch?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1wnyt6p-3 iMUxK"
    >
      <div
        class="accordion__Header-sc-1wnyt6p-4 fDhNEC"
      >
        How do I cancel?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1wnyt6p-3 iMUxK"
    >
      <div
        class="accordion__Header-sc-1wnyt6p-4 fDhNEC"
      >
        What can I watch on Netflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders <Accordion /> with populated data 1`] = `
<div
  class="accordion__Container-sc-1wnyt6p-0 kRFsOa"
>
  <div
    class="accordion__Inner-sc-1wnyt6p-1 ifgAxd"
  >
    <h1
      class="accordion__Title-sc-1wnyt6p-2 dUpgvE"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-sc-1wnyt6p-3 iMUxK"
    >
      <div
        class="accordion__Header-sc-1wnyt6p-4 fDhNEC"
      >
        What is Netflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1wnyt6p-3 iMUxK"
    >
      <div
        class="accordion__Header-sc-1wnyt6p-4 fDhNEC"
      >
        How much does Netflix cost?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1wnyt6p-3 iMUxK"
    >
      <div
        class="accordion__Header-sc-1wnyt6p-4 fDhNEC"
      >
        Where can I watch?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1wnyt6p-3 iMUxK"
    >
      <div
        class="accordion__Header-sc-1wnyt6p-4 fDhNEC"
      >
        How do I cancel?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1wnyt6p-3 iMUxK"
    >
      <div
        class="accordion__Header-sc-1wnyt6p-4 fDhNEC"
      >
        What can I watch on Netflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;
